api/src/main/java/marquez/api/models/Metadata.java

```diff
@@ -0,0 +1,406 @@
+    }
+
+    public Optional<IO> getIo() {
+      return Optional.ofNullable(io);
+    }
+
+    /** ... */
+    static Instant toUtc(@NonNull final ZonedDateTime transitionedOn) {
+      return transitionedOn.withZoneSameInstant(ZoneOffset.UTC).toInstant();
+    }
+  }
+
+  /** ... */
+  @Builder
+  @ToString
+  public static final class Job {
+    @Getter private final JobVersionId versionId;
+    @Getter private final JobType type;
+    @Getter private final JobName name;
+    @Getter private final NamespaceName namespace;
+    @Nullable private final String description;
+    @Nullable private final URI location;
+    @Nullable private final IO io;
+
+    /* ... */
+    public static Job newInstanceWith(
+        @NonNull final OpenLineage.Run run,
+        @NonNull final OpenLineage.Job job,
+        @Nullable final List<OpenLineage.InputDataset> inputs,
+        @Nullable final List<OpenLineage.OutputDataset> outputs) {
+      final NamespaceName namespaceName = NamespaceName.of(job.getNamespace());
+      final JobName jobName = JobName.of(job.getName());
+      final Optional<URI> jobLocation = Facets.locationFor(job);
+      final IO io = IO.newInstanceWith(run, inputs, outputs);
+
+      // ...
+      final JobVersionId jobVersionId =
+          VersionId.forJob(namespaceName, jobName, jobLocation.orElse(null), io);
+
+      // ...
+      return Job.builder()
+          .type(JobType.BATCH)
+          .name(jobName)
+          .namespace(namespaceName)
+          .versionId(jobVersionId)
+          .description(Facets.descriptionFor(job).orElse(null))
+          .location(jobLocation.orElse(null))
+          .io(io)
+          .build();
+    }
+
+    /* ... */
+    public static Job newInstanceFor(@NonNull final OpenLineage.JobEvent event) {
+      final OpenLineage.Job job = event.getJob();
+      final NamespaceName namespaceName = NamespaceName.of(job.getNamespace());
+      final JobName jobName = JobName.of(job.getName());
+      final Optional<URI> jobLocation = Facets.locationFor(job);
+      final IO io = IO.newInstanceWith(event.getInputs(), event.getOutputs());
+
+      // ...
+      final JobVersionId jobVersionId =
+          Utils.newJobVersionWith(namespaceName, jobName, jobLocation.orElse(null), io);
+
+      // ...
+      return Job.builder()
+          .type(JobType.BATCH)
+          .name(jobName)
+          .namespace(namespaceName)
+          .versionId(jobVersionId)
+          .description(Facets.descriptionFor(job).orElse(null))
+          .location(jobLocation.orElse(null))
+          .io(io)
+          .build();
+    }
+
+    public Optional<String> getDescription() {
+      return Optional.ofNullable(description);
+    }
+
+    public Optional<URI> getLocation() {
+      return Optional.ofNullable(location);
+    }
+
+    public Optional<IO> getIo() {
+      return Optional.ofNullable(io);
+    }
+  }
+
+  @Builder
+  @ToString
+  @EqualsAndHashCode
+  public static final class Dataset {
```

Comment:
I think it's worth including column lineage as well as the integration type.

Reply:
Given this PR has been out for a while, and the `Metadata` class isn't yet used within the codebase, I'll add column lineage facet extraction as follow up.